src/fheroes2/editor/editor_object_popup_window.cpp

```diff
@@ -64,7 +66,35 @@ namespace
 
             // This is an invalid object!
             assert( 0 );
-            return "Invalid resource";
+            break;
+        }
+        case MP2::OBJ_ARTIFACT: {
+            for ( const auto & info : Maps::getObjectsByGroup( Maps::ObjectGroup::ADVENTURE_ARTIFACTS ) ) {
+                assert( !info.groundLevelParts.empty() );
+
+                if ( info.objectType == MP2::OBJ_ARTIFACT && info.groundLevelParts.front().icnIndex == tile.getMainObjectPart().icnIndex
+                     && info.groundLevelParts.front().icnType == tile.getMainObjectPart().icnType ) {
+                    return MP2::StringObject( MP2::OBJ_ARTIFACT ) + std::string( " - " ) + Artifact( static_cast<int32_t>( info.metadata[0] ) ).GetName();
+                }
+            }
+
+            // This is an invalid object!
+            assert( 0 );
+            break;
+        }
+        case MP2::OBJ_MONSTER: {
+            for ( const auto & info : Maps::getObjectsByGroup( Maps::ObjectGroup::MONSTERS ) ) {
+                assert( !info.groundLevelParts.empty() );
+
+                if ( info.objectType == MP2::OBJ_MONSTER && info.groundLevelParts.front().icnIndex == tile.getMainObjectPart().icnIndex
+                     && info.groundLevelParts.front().icnType == tile.getMainObjectPart().icnType ) {
+                    return MP2::StringObject( MP2::OBJ_MONSTER ) + std::string( " - " ) + Monster( static_cast<int32_t>( info.metadata[0] ) ).GetMultiName();
```

Comment:
The same comment - a suggestion to print this way: 
![изображение](https://github.com/user-attachments/assets/793919a2-9c5a-4a63-a019-cd6ab00a4066)

```suggestion
                    return MP2::StringObject( MP2::OBJ_MONSTER ) + std::string( "\n" ) + Monster( static_cast<int32_t>( info.metadata[0] ) ).GetMultiName();
```

Currently it is:
![изображение](https://github.com/user-attachments/assets/9c68382b-99de-4273-91a0-83ac411a7afa)